atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/service/TransactionServices.java

```diff
@@ -45,17 +46,24 @@ private static TransactionService createSplitKeyTransactionService(
         return new PreStartHandlingTransactionService(
                 new SplitKeyDelegatingTransactionService<>(
                         transactionSchemaManager::getTransactionsSchemaVersion,
-                        ImmutableMap.of(1, createV1TransactionService(keyValueService))));
+                        ImmutableMap.of(
+                                1, createV1TransactionService(keyValueService),
+                                2, createV2TransactionService(keyValueService))));
```

Comment:
non-urgent: magic numbers should probably be a constant, though I'm not sure where to put them (maybe `TransactionConstants`, or we can try something more clever)